Tests/Realm.Tests/Database/RealmValueTests.cs

```diff
@@ -408,6 +408,61 @@ private void RunNumericTests(RealmValue rv, long value, bool isManaged)
             Assert.That(rv != RealmValue.Null);
         }
 
+        [Test]
+        public void EmbeddedObjectAsRealmValue_WhenAddedToRealm_Throws()
+        {
+            _realm.Write(() =>
+            {
+                var rvo = new RealmValueObject
+                {
+                    RealmValueProperty = new EmbeddedIntPropertyObject()
+                };
+                Assert.Throws<NotSupportedException>(() => _realm.Add(rvo));
```

Comment:
Shouldn't setting `RealmValueProperty` throw an exception directly? When setting the property to an embedded object

Reply:
We discussed internally why it wouldn't be a good idea, so I'm resolving this